core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -9597,6 +9729,13 @@ private boolean isAnInternalRole(String roleName) {
                 .startsWith(WORKFLOW_DOMAIN.toLowerCase());
     }
 
+    private boolean isHybridGroup(String domainName) {
```

Comment:
We already have a method called "isAnInternalRole()" https://github.com/wso2/carbon-kernel/blob/34e69150a4daa6e3f35f1b5e02d687d58c72260a/core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java#L9559

Please check wether it can be re-used instead of introducing a redundant method.

Reply:
In this method it takes the domain from the role name, In a situation where a group is taken using group ID, only domain name is there figure out whether the role is hybrid or not. For those situations this method is introduced